util/system/path.go

```diff
@@ -16,3 +25,180 @@ func DefaultPathEnv(os string) string {
 	}
 	return DefaultPathEnvUnix
 }
+
+// NormalizePath functions similarly to filepath.Join(), but with a few distinctions:
+//   - use the platform flag to decide how we clean the path
+//   - always return an absolute path
+//   - always strip drive letters for Windows paths
+//   - optionally keep the trailing slashes on paths
+//   - paths are returned using forward slashes
+func NormalizePath(parent, path, platform string, keepSlash bool) (string, error) {
```

Comment:
This is not `platform`, it is OS

Reply:
Does ```targetOS``` sound better?

Reply:
It's not really targetOS iiuc. The targetOS of this function is always UNIX. It is like `pathOS` or `inputOS`